### ISSUE TITLE
Null reference exception

### ISSUE THREAD

**Issue by ossaltu** [AngleSharp/AngleSharp#499]
I get null reference exception when I try to load some complex html from stream. Here is the isolated case:

```
var uri = "http://imama.shop.by/kolyaski/detskaya_kolyaska_tutis_zippy_2_v_1_cvet_12_shokoladnyy223222222/";
var config = new Configuration()
     .WithDefaultLoader(s => s.IsResourceLoadingEnabled = true)
     .WithCss();

// this one works
// var document = await BrowsingContext.New(config).OpenAsync(uri);

var req = new HttpClient();
var message = new HttpRequestMessage(System.Net.Http.HttpMethod.Get, uri);

IDocument document1;
using (var response = await req.SendAsync(message))
{
    if (response != null && response.StatusCode == HttpStatusCode.OK)
    {
        var stream = await response.Content.ReadAsStreamAsync();
        // this does not
        document1 = await BrowsingContext.New(config).OpenAsync(m => m.Content(stream).Address(uri));
     }
}
```

**Comment by FlorianRappl** [comment]
I'll investigate, but to me it looks like that the exception has nothing to do with the stream or complex HTML, but rather with the resource loading mechanism from insufficient data.

Thanks for the report 👍  - will be fixed in the next release.

**Comment by FlorianRappl** [comment]
Fixed, see PR #500.
